### PR TITLE
Feat: Bugsnag reporter

### PR DESCRIPTION
**Pull request by nikolajlovenhardt**
Added support for bugsnag.

### CODE REVIEW

File: src/ExceptionHandler.php [hunk 1]
```diff
@@ -6,10 +6,10 @@
 use ReflectionClass;
 use InvalidArgumentException;
 use Asm89\Stack\CorsService;
-use Illuminate\Foundation\Exceptions\Handler as LaravelExceptionHandler;
 use Optimus\Heimdal\Formatters\BaseFormatter;
-use Optimus\Heimdal\Reporters\ReporterInterface;
 use Illuminate\Contracts\Container\Container;
+use Optimus\Heimdal\Reporters\ReporterInterface;
```
Comment (esbenp) [inline src/ExceptionHandler.php:11]: Why is this moved?

File: src/ExceptionHandler.php [hunk 2]
```diff
@@ -6,10 +6,10 @@
...
+use Illuminate\Foundation\Exceptions\Handler as LaravelExceptionHandler;
```
Comment (esbenp) [inline src/ExceptionHandler.php]: Why is this moved?

File: src/ExceptionHandler.php [hunk 3]
```diff
@@ -61,32 +93,52 @@ public function render($request, Exception $e)
...
-            if ($e instanceof $exceptionType) {
-                if (!class_exists($formatter) ||
-                    !(new ReflectionClass($formatter))->isSubclassOf(new ReflectionClass(BaseFormatter::class))) {
-                    throw new InvalidArgumentException("$formatter is not a valid formatter class.");
-                }
-
-                $formatterInstance = new $formatter($this->config, $this->debug);
-                $formatterInstance->format($response, $e, $this->reportResponses);
-                break;
+            if (!$e instanceof $exceptionType) {
```
Comment (esbenp) [inline src/ExceptionHandler.php]: I think you need to do `if (!($e instanceof $exceptionType)) {`

Otherwise a good idea to refactor it like this 👍

**Review by esbenp — CHANGES_REQUESTED** [review]

File: src/ExceptionHandler.php (hunk 3 above)
Comment (nikolajlovenhardt) [reply src/ExceptionHandler.php]: 👍

File: src/ExceptionHandler.php (hunk 1 above)
Comment (nikolajlovenhardt) [inline src/ExceptionHandler.php]: My bad

File: src/ExceptionHandler.php (hunk 2 above)
Comment (nikolajlovenhardt) [inline src/ExceptionHandler.php]: My bad
